engine/api/workflow/workflow_parser.go

```diff
@@ -27,41 +27,35 @@ type ImportOptions struct {
 }
 
 // Parse parse an exportentities.workflow and return the parsed workflow
-func Parse(ctx context.Context, proj sdk.Project, ew *exportentities.Workflow) (*sdk.Workflow, error) {
-	log.Info(ctx, "Parse>> Parse workflow %s in project %s", ew.Name, proj.Key)
+func Parse(ctx context.Context, proj sdk.Project, ew exportentities.Workflow) (*sdk.Workflow, error) {
+	log.Info(ctx, "Parse>> Parse workflow %s in project %s", ew.GetName(), proj.Key)
 	log.Debug("Parse>> Workflow: %+v", ew)
 
-	//Check valid application name
-	rx := sdk.NamePatternRegex
-	if !rx.MatchString(ew.Name) {
-		return nil, sdk.WrapError(sdk.ErrInvalidApplicationPattern, "Workflow name %s do not respect pattern %s", ew.Name, sdk.NamePattern)
-	}
-
-	//Inherit permissions from project
-	if len(ew.Permissions) == 0 {
-		ew.Permissions = make(map[string]int)
-		for _, p := range proj.ProjectGroups {
-			ew.Permissions[p.Group.Name] = p.Permission
-		}
-	}
-
 	//Parse workflow
-	w, errW := ew.GetWorkflow()
+	w, errW := exportentities.ParseWorkflow(ew)
 	if errW != nil {
 		return nil, sdk.NewError(sdk.ErrWrongRequest, errW)
 	}
 	w.ProjectID = proj.ID
 	w.ProjectKey = proj.Key
 
+	// Get permission from project if needed
```

Comment:
why do we need to do it here, i think it is done later when insert the workflow ?

Reply:
Just move code  after parse instead of before parse